lib/sso_cookie_jar.rb

```diff
@@ -1,76 +1,30 @@
 # Creates a separate cookie jar on top of Rails' internal ones
 # https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/middleware/cookies.rb
 class SsoCookieJar < ActionDispatch::Cookies::AbstractCookieJar
-  def initialize(parent_jar)
-    super
-
-    secrets = Rails.application.secrets.sso
-    @signature_private_key = OpenSSL::PKey::RSA.new secrets[:signature_private_key]
-    @signature_public_key = OpenSSL::PKey::RSA.new secrets[:signature_public_key]
-    @signature_algorithm = secrets[:signature_algorithm].to_sym
-
-    @encryption_private_key = OpenSSL::PKey::RSA.new(secrets[:encryption_private_key]) \
-      rescue secrets[:encryption_private_key]
-    @encryption_public_key = OpenSSL::PKey::RSA.new(secrets[:encryption_public_key]) \
-      rescue secrets[:encryption_public_key]
-    @encryption_algorithm = secrets[:encryption_algorithm]
-    @encryption_method = secrets[:encryption_method]
-
-    cookie_secrets = secrets[:cookie]
-    @cookie_name = cookie_secrets[:name]
-    @cookie_options = cookie_secrets[:options]
-  end
+  secrets = Rails.application.secrets.sso[:cookie]
+  @@cookie_name = secrets[:name]
+  @@cookie_options = secrets[:options]
 
   def subject
-    cookie = self[@cookie_name]
+    cookie = self[@@cookie_name]
```

Comment:
Learning something new everyday working with you on this! 
https://stackoverflow.com/questions/5890118/what-does-variable-mean-in-ruby/5890199#5890199

Reply:
Normally class variables are bad if you are going to modify them (they are not thread-safe and propagate to all descendants I think) but here we don't modify it and I just need them to be available in the instance methods so they are fine